services/venice-controller/src/main/java/com/linkedin/venice/controller/logcompaction/CompactionManager.java

```diff
@@ -0,0 +1,123 @@
+   * @param clusterName
+   * @param childControllers
+   * @return list of StoreInfo of stores ready for log compaction in clusterName
+   */
+  public List<StoreInfo> getStoresForCompaction(String clusterName, Map<String, ControllerClient> childControllers) {
+    ArrayList<StoreInfo> storeInfoList = new ArrayList<>();
+
+    // iterate through child controllers
+    for (Map.Entry<String, ControllerClient> controller: childControllers.entrySet()) {
+
+      // add all store info to storeInfoList
+      MultiStoreInfoResponse response = controller.getValue().getClusterStores(clusterName);
+      storeInfoList.addAll(response.getStoreInfoList());
+    }
+
+    // filter for stores ready for log compaction
+    return filterStoresForCompaction(storeInfoList);
+  }
+
+  // public for testing
+  List<StoreInfo> filterStoresForCompaction(ArrayList<StoreInfo> storeInfoList) {
+    ArrayList<StoreInfo> compactionReadyStores = new ArrayList<>();
+    for (StoreInfo storeInfo: storeInfoList) {
+      if (isCompactionReady(storeInfo)) {
+        compactionReadyStores.add(storeInfo);
+      }
+    }
+    return compactionReadyStores;
+  }
+
+  // This function abstracts the criteria for a store to be ready for compaction
+  private boolean isCompactionReady(StoreInfo storeInfo) {
+    boolean isHybridStore = storeInfo.getHybridStoreConfig() != null;
+
+    return isHybridStore && isLastCompactionTimeOlderThanThresholdHours(DEFAULT_COMPACTION_THRESHOLD_HOURS, storeInfo);
+    // TODO LC: get compaction threshold hours from config
+  }
+
+  /**
+   * This function checks if the last compaction time is older than the threshold.
+   * @param compactionThresholdHours, the number of hours that the last compaction time should be older than
+   * @param storeInfo, the store to check the last compaction time for
+   * @return true if the last compaction time is older than the threshold, false otherwise
+   */
+  private boolean isLastCompactionTimeOlderThanThresholdHours(int compactionThresholdHours, StoreInfo storeInfo) {
+    // get the last compaction time
+    int currentVersionNumber = storeInfo.getCurrentVersion();
+    Optional<Version> currentVersion = storeInfo.getVersion(currentVersionNumber);
+    if (!currentVersion.isPresent()) {
+      LOGGER.warn("Couldn't find current version: {} from store: {}", currentVersionNumber, storeInfo.getName());
+      return false; // invalid store because no current version, this store is not eligible for compaction
+    }
+
+    // calculate hours since last compaction
+    long lastCompactionTime = currentVersion.get().getCreatedTime();
+    long currentTime = System.currentTimeMillis();
+    long millisecondsSinceLastCompaction = currentTime - lastCompactionTime;
+    long hoursSinceLastCompaction = TimeUnit.MILLISECONDS.toHours(millisecondsSinceLastCompaction);
+
+    return hoursSinceLastCompaction > compactionThresholdHours;
```

Comment:
if you are a big fan of lambda or Optional, this can be re-written in 
```java
return storeInfo.getVersion(currentVersionNumber)
            .map(v -> {
                 // calculate hours since last compaction
                 long lastCompactionTime = v.get().getCreatedTime();
                 long currentTime = System.currentTimeMillis();
                 long millisecondsSinceLastCompaction = currentTime - lastCompactionTime;
                 long hoursSinceLastCompaction = 
                 TimeUnit.MILLISECONDS.toHours(millisecondsSinceLastCompaction);
  
                  return hoursSinceLastCompaction > compactionThresholdHours;
             ).orElse (false);
```

Reply:
elegant! implemented.